File: node_mongo/app_api/models/reviews.js
const mongoose = require('mongoose');
const ObjectId = mongoose.Schema.Types.ObjectId;

const reviewSchema = new mongoose.Schema({
    book: {
        type: ObjectId,
        required: true
    },
    reviewer: {
        type: ObjectId,
        required: true
    },
    body: {
        type: String,
        required: true
    },
    start: Date,
    stop: Date
});

mongoose.model('Review', reviewSchema);

File: rails_mysql/spec/factories/reviews.rb
FactoryBot.define do
  factory :review do
    book
    reviewer
    body { Faker::Markdown.random }

    factory :review_in_progress do
      start { Faker::Date.backward }

      factory :completed_review do
        stop { Faker::Date.between from: start, to: Date.today }
      end
    end
  end
end
